src/mbgl/map/map.cpp

```diff
@@ -43,22 +43,31 @@ void Map::resume() {
     paused = false;
 }
 
+void Map::setView(View* view_) {
+    view = view_;
+    transform = std::make_unique<Transform>(*view);
+    context->invoke(&MapContext::setView, view);
```

Comment:
You might need to make this a `context->invokeSync` to make sure it is safe to reuse the old view (i.e. `deactivate()` was called) in another map object right after calling this method.